crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -522,28 +621,88 @@ impl<'a> DependencyProvider for CondaDependencyProvider<'a> {
+                        &mut parse_match_spec_cache,
+                    ) {
+                        Ok(version_set_id) => version_set_id,
+                        Err(e) => {
+                            let reason = self.pool.intern_string(format!(
+                                "the optional dependency '{req}' for feature '{feature_name}' failed to parse: {e}"
+                            ));
+                            return Dependencies::Unknown(reason);
+                        }
+                    };
+                    for version_set_id in version_set_id {
+                        dependencies.requirements.push(version_set_id.into());
                     }
+                }
+
+                // Add a dependency back to the base package with exact version
+                let base_spec = MatchSpec {
+                    name: Some(record.package_record.name.clone()),
+                    version: Some(VersionSpec::Exact(
+                        EqualityOperator::Equals,
+                        record.package_record.version.version().clone(),
+                    )),
+                    build: Some(StringMatcher::Exact(record.package_record.build.clone())),
+                    build_number: Some(BuildNumberSpec::new(
+                        OrdOperator::Eq,
+                        record.package_record.build_number,
+                    )),
+                    subdir: Some(record.package_record.subdir.clone()),
+                    md5: record.package_record.md5,
+                    sha256: record.package_record.sha256,
+                    ..Default::default()
                 };
 
-            dependencies.requirements.push(version_set_id.into());
+                let (name, nameless_spec) = base_spec.into_nameless();
+                let name_id = self.pool.intern_package_name(
+                    name.expect("cannot use matchspec without a name")
+                        .as_normalized(),
+                );
+                let version_set_id = self.pool.intern_version_set(name_id, nameless_spec.into());
+                dependencies.requirements.push(version_set_id.into());
+            }
+        } else {
+            // Add regular dependencies
+            for depends in record.package_record.depends.iter() {
+                let version_set_id =
+                    match parse_match_spec(&self.pool, depends, &mut parse_match_spec_cache) {
+                        Ok(version_set_id) => version_set_id,
+                        Err(e) => {
+                            let reason = self.pool.intern_string(format!(
+                                "the dependency '{depends}' failed to parse: {e}",
+                            ));
+
+                            return Dependencies::Unknown(reason);
+                        }
+                    };
+
+                for version_set_id in version_set_id {
+                    dependencies.requirements.push(version_set_id.into());
+                }
```

Comment:
Might be better to use `extend` to ensure we dont push individual items and allocate multiple times.